### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by ekimcoskun** [facebook/docusaurus#10696] [Q&A]
I want to add a token input for my private routes so, the users can enter their token for access. Token will be added to header. Is there any way to do that in current version ?

**Comment by slorber** [comment]
You can find many previous Docusaurus + Authentication issues using search, on which I provided advise.

Your request is not clear enough for me to help you, but I probably answered in other places already.

Docusaurus is a static site generator. If you want static pages to look different for each users, then it's not the right tool. 

Now you can have pages look different for each user by using client-side JS/React, and the `<BrowserOnly>` component. In this case, the static page will contain a placeholder/loader, and the real, authenticated data will be loaded after. We have an example on our own website here: https://docusaurus.io/feature-requests/p/update-mermaid-version-in-theme-mermaid

If you want to protect your whole doc site, you can use the "middleware" feature of CDNs to gate access. You can do so for your whole site, or a subset of it. But keep in mind protecting a subset is not a perfect solution, because client-site JS files in `website/build/assets/js` remain accessible and may contain sensitive data. Similarly, if you want different users to see a different UI (for example having their profile picture in the navbar), this must be done with client-side JS.
